framework/console/controllers/CacheController.php

```diff
@@ -270,6 +270,12 @@ private function findCaches(array $cachesNames = [])
                 $caches[$name] = $component['class'];
             } elseif (is_string($component) && $this->isCacheClass($component)) {
                 $caches[$name] = $component;
+            } elseif (is_callable($component)) {
```

Comment:
better check for `instanceof Closure`, callable can also be true for arrays and strings.